core/thread/each_caller_location_spec.rb

```diff
@@ -0,0 +1,81 @@
+require_relative '../../spec_helper'
+
+describe "Thread.each_caller_location" do
+  ruby_version_is "3.2" do
+    it "iterates through the current execution stack and matches caller_locations content and type" do
+      ar = []
+      Thread.each_caller_location { ar << _1; }
```

Comment:
**suggestion:** it makes sense to use only basic limited syntax in specs to allow new Ruby implementations to run a test (or just to load a file without `SyntaxError`) even if it doesn't support some features (e.g. numbered parameters).

Right now numbered parameters are used in RubySpecs only in `/language/numbered_parameters_spec.rb` to it make sense to still avoid using them.

Reply:
**suggestion:** Also it's convenient to use `ScratchPad` to accumulate result.